x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/editor_toggle_controls.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiButton, EuiText } from '@elastic/eui';
+
+import { useDispatch, useState } from '../../mappings_state';
+import { FieldsEditor } from '../../types';
+import { canUseMappingsEditor, normalize } from '../../lib';
+
+interface Props {
+  editor: FieldsEditor;
+}
+
+/* TODO: Review toggle controls UI */
+export const EditorToggleControls = ({ editor }: Props) => {
+  const dispatch = useDispatch();
+  const state = useState();
+
+  const [showMaxDepthWarning, setShowMaxDepthWarning] = React.useState<boolean>(false);
+  const [showValidityWarning, setShowValidityWarning] = React.useState<boolean>(false);
+
+  const clearWarnings = () => {
+    if (showMaxDepthWarning) setShowMaxDepthWarning(false);
+    if (showValidityWarning) setShowValidityWarning(false);
+  };
+
+  if (editor === 'default') {
+    clearWarnings();
+    return (
+      <EuiButton
+        onClick={() => {
+          dispatch({ type: 'documentField.changeEditor', value: 'json' });
+        }}
+      >
+        Use JSON Editor
+      </EuiButton>
+    );
+  }
+
+  return (
+    <>
+      <EuiButton
+        onClick={() => {
+          clearWarnings();
```

Comment:
Ah ok, I think I see better what you meant regarding inlining. This maybe comes down more to personal preference for me, but I think this reads a bit better since the one handler is so small.